data/json/items/gun/223.json

```diff
@@ -292,13 +196,89 @@
       }
     ]
   },
+  {
+    "//": "First group change from 'NATO Assault Rifle'. Groups all 'H&K416' related rifles together, alongside the SCAR-L due to its similar properties. Used M27 due to it being the most popular of the four in the US.",
+    "id": "m27_assault_rifle",
+    "copy-from": "rifle_auto",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "M27 Assault Rifle" },
+    "description": "Designed to replace the M4 Carbine, the M27 Assault Rifle features most of the former's strengths, while being considerably more durable.",
+    "weight": "3490 g",
+    "volume": "4957 ml",
+    "longest_side": "798 mm",
+    "price": 540000,
+    "price_postapoc": 2250,
+    "variants": [
+      {
+        "id": "h&k416a5",
+        "name": { "str": "H&K416 A5" },
+        "description": "Designed to replace the M4A1, the Heckler and Koch 416A5 features most of the former's strengths, while being considerably more durable."
+      },
+      {
+        "id": "m27iar",
+        "name": { "str": "M27 IAR" },
+        "description": "A H&K416 carbine outfitted with a heavier barrel to enable higher amounts of suppressive fire while retaining a good degree of mobility.",
+        "weight": 1
```

Comment:
If you have an idea of what the distribution of these various variants should be, you can add weights to the others ones. I just do it like this because I don't know.

Reply:
Oh ok. I added in the weight to it because otherwise it didn't work with the debug menu very nicely while I was originally testing.

Reply:
What happens is that when these items are spawned without a variant (such as through the debug menu), the game does a raffle among the variants, selecting a variant according to its weight. If none of them have a weight, the item will spawn without a variant (until a save and reload), and thus appear as the generic version.

Reply:
Alright, not completely sure on what weights should be given to the other rifles, so I'll leave it for now